pkg/analyzer/scan.go

```diff
@@ -110,17 +113,65 @@ func ScanK8sServiceObject(kind string, objDataBuf []byte) (*common.Service, erro
 	return &serviceCtx, nil
 }
 
-// Create an OpenShift Route object from a buffer
-func ScanOCRouteObject(kind string, objDataBuf []byte) (*ocroutev1.Route, error) {
+// Scan an OpenShift Route object and mark the services it uses to be exposed inside the cluster
+func ScanOCRouteObject(kind string, objDataBuf []byte, servicesToExpose common.ServicesToExpose) error {
 	if kind != "Route" {
-		return nil, fmt.Errorf("expected parsing a Route resource, but got `%s`", kind)
+		return fmt.Errorf("expected parsing a Route resource, but got `%s`", kind)
 	}
 
-	routeObj := parseRoute(bytes.NewReader(objDataBuf))
+	routeObj := parseResource[ocroutev1.Route](bytes.NewReader(objDataBuf))
 	if routeObj == nil {
-		return nil, fmt.Errorf("failed to parse Route resource")
+		return fmt.Errorf("failed to parse Route resource")
 	}
-	return routeObj, nil
+
+	exposedServicesInNamespace, ok := servicesToExpose[routeObj.Namespace]
+	if !ok {
+		servicesToExpose[routeObj.Namespace] = map[string]bool{}
```

Comment:
does it handle Ingress/Route resources without a namespace?

Reply:
I believe it does. For such resources `Namespace` field is `""`, which is a valid key in the map.
Thus, it will only expose listed services which also do not have Namespace.
The BookInfo example has such an Ingress resource, and a corresponding Service. It seems to be exposed properly.